rules/api/rules.py

```diff
@@ -251,3 +252,12 @@ def delete(self, request, *args, **kwargs):
             DELETE /api/proxy/rules/app/ HTTP/1.1
         """
         return super(RuleDetailAPIView, self).delete(request, *args, **kwargs)
+
+
+class UserEngagementAPIView(ListAPIView):
+
+    serializer_class = UserEngagementSerializer
+
+    def get_queryset(self):
+        User = get_user_model()
+        return User.objects.order_by('-last_login').prefetch_related('engagements')
```

Comment:
Check the actual SQL that is generated here. `prefetch_related` is usually blowing up queries that could otherwise be very simple.

Reply:
I’ve checked already with built in logger - 2 queries.